### PR TITLE
Export to hf

### PR DESCRIPTION
**Pull request by pszemraj**
- remaining cleanup from previous GLUE pr
- export a pretrained checkpoint to same format as https://huggingface.co/chandar-lab/NeoBERT

### CODE REVIEW

**Comment by pszemraj** [comment]
short-term items to resolve:

- [ ] MLM language head compatibility + export to hf transformers, **with retaining weights**. should make rasonable predictions in ad-hoc tests
  - [ ] double-check `scripts/export-hf/validate.py` and ensure it doesn't have a bunch of cc-style warning/error supression to say everything works
- [ ] .yaml pretrain config saved "inside" the generated readme does not respect/format tabs correctly
- [ ] do not save pointless `training_info` key to `config.json` ([example](https://huggingface.co/BEE-spoke-data/neobert-100k-test/blob/main/config.json#L24))

**Comment by pszemraj** [comment]
will retrain and report back:

##  MLM language head compatibility Bug

In `src/neobert/pretraining/trainer.py` (line 585), the model was saved with:

```python
torch.save(model.state_dict(), ...)  # WRONG — saves wrapped model
```

It should have been:

```python
torch.save(accelerator.unwrap_model(model).state_dict(), ...)  # CORRECT
```

---

## Why This Caused MLM Failure

1. During training, the live model worked fine (65% MLM accuracy).
2. When saving, the wrapped model’s state was saved instead of the actual model.
3. Upon loading, the corrupted weights caused:

   * Encoder outputs with extreme variance (std = 3.6 instead of \~1.0).
   * RMSNorm weights amplified to **3.3×** normal.
   * Decoder always predicting token **24 ("1")**.

---

## Evidence Trail

* Training metrics showed **65% MLM accuracy** ✓
* GLUE scores were excellent (**85–90%**) ✓
* But MLM inference always predicted **"1"** ✗
* Encoder output std was **3.6** (should be \~1.0).
* RMSNorm weights mean was **3.33** (should be \~1.0).

**Comment by pszemraj** [comment]
Turns out the real issue was the method of running "mlm prediction" and the custom tokenizer which always pretends metaspace. 

But the save improvements are still valid and work, so will keep those. Finally good to go
